src/sentry/static/sentry/app/views/projectDetail/projectLatestAlerts.tsx

```diff
@@ -37,19 +39,36 @@ type State = {
 
 class ProjectLatestAlerts extends AsyncComponent<Props, State> {
   shouldComponentUpdate(nextProps: Props, nextState: State) {
+    const {location, isProjectStabilized} = this.props;
     // TODO(project-detail): we temporarily removed refetching based on timeselector
     if (
       this.state !== nextState ||
-      this.props.location.query.environment !== nextProps.location.query.environment
+      didProjectOrEnvironmentChange(location, nextProps.location) ||
+      isProjectStabilized !== nextProps.isProjectStabilized
     ) {
       return true;
     }
 
     return false;
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const {location, isProjectStabilized} = this.props;
+
+    if (
+      didProjectOrEnvironmentChange(prevProps.location, location) ||
+      prevProps.isProjectStabilized !== isProjectStabilized
```

Comment:
should we include `isProjectStabilized` in the function too?

Reply:
I'd rather not - let's keep it more explicit.